aria_data_loaders/aria_data_utils/spot_aria_p1_node.py

```diff
@@ -0,0 +1,591 @@
+        self,
+        spot_frame: str = "vision",
+        data_size_for_avg: int = 10,
+        filter_dist: float = 2.2,
+    ):
+        """
+        Returns the average transformation of spot world frame to marker frame
+
+        We get a camera_T_marker for each frame in which marker is detected.
+        Depending on the frame rate of image capture, multiple frames may have captured the marker.
+        Averaging all transforms would be best way to compensate for any noise that may exist in any frame's detections
+        camera_T_marker is used to compute spot_T_marker[i] and thus spotWorld_T_marker[i].
+        Then we average all spotWorld_T-marker to find average marker pose wrt spotWorld.
+        """
+        if spot_frame != "vision" and spot_frame != "odom":
+            raise ValueError("base_frame should be either vision or odom")
+        cv2.namedWindow("hand_image", cv2.WINDOW_AUTOSIZE)
+        spot_world_frame = spot_frame
+
+        # Get Hand camera intrinsics
+        hand_cam_intrinsics = self.spot.get_camera_intrinsics(SpotCamIds.HAND_COLOR)
+        hand_cam_intrinsics = self._to_camera_metadata_dict(hand_cam_intrinsics)
+        hand_cam_pose_estimator = AprilTagPoseEstimator(hand_cam_intrinsics)
+
+        # Register marker ids
+        marker_ids_list = [DOCK_ID]
+        # marker_ids_list = [i for i in range(521, 550)]
+        hand_cam_pose_estimator.register_marker_ids(marker_ids_list)
+
+        marker_position_from_dock_list = []  # type: List[Any]
+        marker_quaternion_form_dock_list = []  # type: List[Any]
+
+        marker_position_from_robot_list = []  # type: List[Any]
+        marker_quaternion_form_robot_list = []  # type: List[Any]
+
+        while len(marker_position_from_dock_list) < data_size_for_avg:
+            print(f"Iterating - {len(marker_position_from_dock_list)}")
+            is_marker_detected_from_hand_cam = False
+            img_response_hand = self.spot.get_hand_image()
+            img_hand = image_response_to_cv2(img_response_hand)
+
+            (
+                img_rend_hand,
+                hand_sp_handcam_T_marker,
+            ) = hand_cam_pose_estimator.detect_markers_and_estimate_pose(
+                img_hand, should_render=True
+            )
+
+            if hand_sp_handcam_T_marker is not None:
+                print("Trackedddd")
+                is_marker_detected_from_hand_cam = True
+                hand_mn_handcam_T_marker = (
+                    Spot.convert_transformation_from_sophus_to_magnum(
+                        hand_sp_handcam_T_marker
+                    )
+                )
+
+            # Spot - spotWorld_T_handcam computation
+            frame_tree_snapshot_hand = img_response_hand.shot.transforms_snapshot
+            hand_mn_body_T_handcam = self._get_body_T_handcam(frame_tree_snapshot_hand)
+            hand_mn_spotWorld_T_handcam = self._get_spotWorld_T_handcam(
+                frame_tree_snapshot_hand, spot_frame=spot_frame
```

Comment:
utilize the generic function to get these transforms

Reply:
This is already a part of issue https://github.com/facebookresearch/spot-sim2real/issues/90. Will be resolved in the PR to solve that issue